src/main/scala/za/co/absa/ultet/model/DBAppModel.scala

```diff
@@ -32,39 +33,79 @@ import za.co.absa.ultet.util.parsers.{PgFunctionFileParser, PgTableFileParser}
 
 import java.net.URI
 
-case class DBAppModel(databases: DatabaseDefs) {
+case class DBAppModel private[model](databases: DatabaseDefs, schemaOwners: SchemaOwners) {
 
   def +(other: Option[DBAppModel]): DBAppModel = {
     other.fold(this)(this + _)
   }
 
   def +(other: DBAppModel): DBAppModel = {
-    val newDatabases = other.databases.foldLeft(databases) {
+
+    def assignNewSchemaOwners(toModel: DBAppModel, schemas: Set[SchemaName], schemaOwnersToApply: SchemaOwners): DBAppModel = {
+      schemas.foldLeft(toModel) {
+        case (acc, schemaName) =>
+          acc.addSchemaOwner(schemaName, schemaOwnersToApply(schemaName))
+      }
+    }
+    // merge owners
+    val thisSchemasWithOwners = schemaOwners.keys.toSet
+    val otherSchemasWithOwners = other.schemaOwners.keys.toSet
+
+    val thisWithNewOwners = assignNewSchemaOwners(this, otherSchemasWithOwners.diff(thisSchemasWithOwners), other.schemaOwners)
+    val otherWithNewOwners = assignNewSchemaOwners(other, thisSchemasWithOwners.diff(otherSchemasWithOwners), schemaOwners)
+
+    val mergedSchemaOwners = other.schemaOwners ++ schemaOwners
+
+    // merge databases
+    val newDatabases = otherWithNewOwners.databases.foldLeft(thisWithNewOwners.databases) {
       case (acc, (dbName, dbDef)) =>
         val newDbDef = dbDef + acc.get(dbName)
         acc + (dbName -> newDbDef)
     }
-    DBAppModel(newDatabases)
+
+    DBAppModel(newDatabases, mergedSchemaOwners)
   }
 
   def addFunctionSource(schemaName: SchemaName, functionSource: String): DBAppModel = {
```

Comment:
add TODO that functions don't yet belong to schemas?

Reply:
i.e. schemaName not used

Reply:
Isn't the lin75 enough?

Reply:
Hmm but what I would expect is that `PgFunctionFileParser` should accept `schemaName` in the constructor